app/services/make_an_offer.rb

```diff
@@ -36,6 +47,48 @@ def save
 
   attr_reader :application_choice
 
+  def offered_course_option
+    return nil if @course_data.nil?
+
+    @course_option
+  end
+
+  def validate_course_data
+    return if @course_data.nil?
```

Comment:
idea:

1. load C_o in the constructor using a method which returns a validatable thing
1. ask here whether that thing is valid, if not assign its errors to ourself